src/nodes.coffee

```diff
@@ -1220,21 +1220,154 @@ exports.Class = class Class extends Base
+      for subclause in @clause
+        code = code.concat subclause.compileNode o
+
+    if @source?.value?
+      code.push @makeCode ' from ' unless @clause is null
+      code.push @makeCode @source.value
+
+    code.push @makeCode ';'
+    code
+
+exports.ExportDeclaration = class ExportDeclaration extends ModuleDeclaration
+  compileNode: (o) ->
+    @checkScope o, 'export'
+
+    code = []
+    code.push @makeCode "#{@tab}export "
+    code.push @makeCode 'default ' if @ instanceof ExportDefaultDeclaration
+
+    if @ not instanceof ExportDefaultDeclaration and
+       (@clause instanceof Assign or @clause instanceof Class)
+      # When the ES2015 `class` keyword is supported, don’t add a `var` here
+      code.push @makeCode 'var '
+      @clause.moduleDeclaration = 'export'
+
+    if @clause.body? and @clause.body instanceof Block
+      code = code.concat @clause.compileToFragments o, LEVEL_TOP
+    else
+      code = code.concat @clause.compileNode o
+
+    code.push @makeCode " from #{@source.value}" if @source?.value?
+    code.push @makeCode ';'
+    code
+
+exports.ExportNamedDeclaration = class ExportNamedDeclaration extends ExportDeclaration
+
+exports.ExportDefaultDeclaration = class ExportDefaultDeclaration extends ExportDeclaration
+
+exports.ExportAllDeclaration = class ExportAllDeclaration extends ExportDeclaration
+
+exports.ModuleSpecifierList = class ModuleSpecifierList extends Base
+  constructor: (@specifiers) ->
+
+  children: ['specifiers']
+
+  compileNode: (o) ->
+    code = []
+    o.indent += TAB
+    compiledList = (specifier.compileToFragments o, LEVEL_LIST for specifier in @specifiers)
+
+    code.push @makeCode("{\n#{o.indent}")
+
+    for fragments, index in compiledList
+      code.push @makeCode(",\n#{o.indent}") if index
+      code.push fragments...
+
+    code.push @makeCode("\n}")
+    code
+
+exports.ImportSpecifierList = class ImportSpecifierList extends ModuleSpecifierList
+
+exports.ExportSpecifierList = class ExportSpecifierList extends ModuleSpecifierList
+
+exports.ModuleSpecifier = class ModuleSpecifier extends Base
+  constructor: (@original, @alias, @moduleDeclarationType) ->
+    # The name of the variable entering the local scope
+    @identifier = if @alias? then @alias.value else @original.value
+
+  children: ['original', 'alias']
+
+  compileNode: (o) ->
+    o.scope.add @identifier, @moduleDeclarationType
+    code = []
+    code.push @makeCode @original.value
+    code.push @makeCode " as #{@alias.value}" if @alias?
+    code
+
+exports.ImportSpecifier = class ImportSpecifier extends ModuleSpecifier
+  constructor: (imported, local) ->
+    super imported, local, 'import'
+
+  compileNode: (o) ->
+    # Per the spec, symbols can’t be imported multiple times
+    # (e.g. `import { foo, foo } from 'lib'` is invalid)
+    if @identifier in o.importedSymbols
+      @error 'already imported'
+    else if o.scope.check(@identifier)
+      @error 'duplicate declaration'
```

Comment:
> You mean just rename "already imported" to "duplicate declaration"?

@GeoffreyBooth I'm answering your question here :) I'm thinking of collapsing the above lines to:

```coffee
if @identifier in o.importedSymbols or o.scope.check(@identifier)
  @error 'duplicate declaration'
```